src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -984,6 +1073,40 @@ RewriteResponse TheoryStringsRewriter::postRewrite(TNode node) {
   {
     retNode = rewriteStringCode(node);
   }
+  else if (node.getKind() == REGEXP_CONCAT)
+  {
+    retNode = rewriteConcatRegExp(node);
+  }
+  else if (node.getKind() == REGEXP_UNION || node.getKind() == REGEXP_INTER)
+  {
+    retNode = rewriteAndOrRegExp(node);
+  }
+  else if (node.getKind() == REGEXP_STAR)
+  {
+    retNode = rewriteStarRegExp(node);
+  }
+  else if (node.getKind() == REGEXP_PLUS)
+  {
+    retNode =
+        nm->mkNode(REGEXP_CONCAT, node[0], nm->mkNode(REGEXP_STAR, node[0]));
```

Comment:
Since we are only doing this post-rewrite, we might miss rewrites of the `nm->mkNode(REGEXP_STAR, node[0])` part. We'd either to return REWRITE_AGAIN or do this as a prerewrite.

Reply:
I think we are calling REWRITE_AGAIN_FULL actually on https://github.com/CVC4/CVC4/blob/master/src/theory/strings/theory_strings_rewriter.cpp#L992, so this should be fine.

Reply:
Hm, ok. Seems, rather inefficient to call it over-and-over again but I guess it is fine for now.